File: Languages/JavaScript/Day022.js
// Languages - JavaScript - Day 22
// Foundational Concepts: Basics, syntax, loops, conditionals, functions, and variables.

// Create a function that removes all duplicate characters from a string while preserving the original order.


// Function to remove duplicates from a string
function removeDuplicates(input) {
    if (!input) {
        return "Input cannot be empty.";
    }
    const seen = new Set();
    let result = '';
    for (const char of input) {
        if (!seen.has(char)) {
            seen.add(char);
            result += char;
        }
    }
    return result;
}

// Get user input and display the result
const input = prompt("Enter a string:");
if (input === null || input.trim() === "") {
    console.log("Input cannot be empty.");
} else {
    const result = removeDuplicates(input);
    console.log("String after removing duplicates:", result);
}

File: Languages/Rust/Day022.rs
// Languages - Rust - Day 22
// Foundational Concepts: Basics, syntax, loops, conditionals, functions, and variables.

// Create a function that removes all duplicate characters from a string while preserving the original order.


use std::collections::HashSet;
use std::io;

// Function to remove duplicates from a string while preserving the original order
fn remove_duplicates(input: &str) -> String {
    if input.trim().is_empty() {
        return String::from("Input cannot be empty.");
    }
    let mut seen = HashSet::new();
    input.chars().filter(|c| seen.insert(*c)).collect()
}

// Get user input and display the result
fn main() {
    println!("Enter a string:");
    let mut input = String::new();
    
    io::stdin()
        .read_line(&mut input)
        .expect("Failed to read line");
    
    let input = input.trim();
    
    if input.is_empty() {
        println!("Input cannot be empty.");
    } else {
        let result = remove_duplicates(input);
        println!("String after removing duplicates: {}", result);
    }
}
